source/base/data_out_base.cc

```diff
@@ -7741,9 +7744,37 @@ DataOutInterface<dim, spacedim>::write_vtu_in_parallel(
                                   vtk_flags,
                                   ss);
 
-    ierr = MPI_File_write_ordered(
-      fh, ss.str().c_str(), ss.str().size(), MPI_CHAR, MPI_STATUS_IGNORE);
+    // using prefix sum to find specific offset to write at.
+    const std::uint64_t size_on_proc = ss.str().size();
+    std::uint64_t prefix_sum = 0;
+    ierr = MPI_Exscan(&size_on_proc, &prefix_sum, 1, MPI_CHAR, MPI_SUM, comm);
+    AssertThrowMPI(ierr);
+
+    // Boradcasting the header size to all processor.
+    ierr = MPI_Bcast(&header_size, 1, MPI_UNSIGNED, 0, comm);
+    AssertThrowMPI(ierr);
+
+    // Locate specific offset for each processor.
+    offset = static_cast<MPI_Offset>(header_size) + prefix_sum;
+
+    ierr = MPI_File_write_at_all(fh,
+                                 offset,
+                                 ss.str().c_str(),
+                                 ss.str().size(),
+                                 MPI_CHAR,
+                                 MPI_STATUS_IGNORE);
     AssertThrowMPI(ierr);
+
+    // Sending the size_on_proc and offset of last rank to rank 0 to write
```

Comment:
I would create a new variable set to offset+size_on_proc on the last rank and only send that to rank 0. For that I would use std::uint64.